prprocessor/__main__.py

```diff
@@ -92,6 +110,31 @@ def summarize(summary: Mapping[str, Iterable], show_headers: bool) -> Generator[
                 yield f'* {line}'
 
 
+async def update_pr_labels(pull_request: Mapping, labels_to_add: Iterable[Label],
+                           labels_to_remove: Iterable[Label]) -> None:
+    github_api = RUNTIME_CONTEXT.app_installation_client
+
+    tasks = []
+
+    repository = pull_request['base']['repo']['full_name']
+
+    # TODO: pull_request['labels_url']
+    # https://github.com/orgs/community/discussions/66499
+    url = f'{pull_request["issue_url"]}/labels{{/name}}'
+
+    if labels_to_add:
+        logger.info('%s PR #%s: adding labels: %r', repository, pull_request['number'], labels_to_add)
+        data = [label.value for label in labels_to_add]
+        tasks.append(github_api.post(url, data=data))
+
+    for label in labels_to_remove:
+        logger.info('%s PR #%s: removing label: %s', repository, pull_request['number'], label)
+        tasks.append(github_api.delete(url, url_vars={'name': label.value}))
```

Comment:
I chose an add + remove instead of put so there wouldn't be race conditions with other (unmanaged) labels. I don't know if this is best, but time will tell. With https://github.com/ekohl/gh-test-env/pull/7#event-10319522098 I saw that the 2 actions weren't squashed to a single line, so it may be more verbose.

Perhaps take a look in production and see what works.